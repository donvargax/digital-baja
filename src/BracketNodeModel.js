import { NodeModel } from "storm-react-diagrams";

import { BracketPortModel } from "./BracketPortModel";

export class BracketNodeModel extends NodeModel {
	constructor(home = "None", visitor = "None", color = "rgb(0,192,255)", played = false, homeWinner = false) {
		super("bracket")
		this.home = home;
		this.visitor = visitor;
		this.color = color;
		this.played = played;
		this.homeWinner = homeWinner;
		this.addPort(new BracketPortModel("left"))
		this.addPort(new BracketPortModel("right"))
	}

	getLeftPort() {
		return this.ports.left
	}

	getRightPort() {
		return this.ports.right
	}

}