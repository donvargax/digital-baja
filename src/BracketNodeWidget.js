import * as _ from "lodash";
import * as React from "react";
import { AbstractNodeFactory, AbstractPortFactory, BaseWidget, DefaultLinkModel, DefaultNodeModel, DiagramEngine, DiagramModel, DiagramWidget, NodeModel, PortModel, PortWidget } from "storm-react-diagrams";
import "storm-react-diagrams/dist/style.min.css";

import './Bracket.css'

export class BracketNodeWidget extends BaseWidget {
	static defaultProps = {
		size: 150,
		node: null
	};

	constructor(props) {
		super("bracket-node", props);
		this.state = {};
	}

	render() {
		return (
			<div {...this.getProps()}
				style={{
					background: this.props.node.color
				}}
			>
				<div className={this.bem("__title")}>
					<div className={this.bem("__name")} style={{ background: this.props.node.played && this.props.node.homeWinner ? "#0a0" : "inherit" }}>{this.props.node.home}</div>
				</div>
				<PortWidget name="left" node={this.props.node} style={{ width: 0, height: 0 }} />
				<PortWidget name="right" node={this.props.node} />
				<div className={this.bem("__title")}>
					<div className={this.bem("__name")} style={{ background: this.props.node.played && !this.props.node.homeWinner ? "#0a0" : "inherit" }} >{this.props.node.visitor}</div>
				</div>
			</div>
		);
	}

}