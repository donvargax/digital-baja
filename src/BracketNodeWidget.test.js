import React from 'react';
import { render } from 'react-testing-library'

import { Toolkit } from "storm-react-diagrams"

import { BracketNodeModel } from './BracketNodeModel';
import { BracketNodeWidget } from './BracketNodeWidget'

Toolkit.TESTING = true;
window.focus = function() { };

it('renders correctly', () => {
    console.log(process.env.NODE_ENV)
    const nodeModel = new BracketNodeModel()
    const { asFragment } = render(<BracketNodeWidget node={nodeModel} />)
    expect(asFragment()).toMatchSnapshot()
});
