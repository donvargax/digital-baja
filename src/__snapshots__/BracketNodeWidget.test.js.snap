// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bracket-node "
    style="background: rgb(0, 192, 255);"
  >
    <div
      class="bracket-node__title "
    >
      <div
        class="bracket-node__name "
      >
        None
      </div>
    </div>
    <div
      class="port srd-port "
      data-name="left"
      data-nodeid="1"
    />
    <div
      class="port srd-port "
      data-name="right"
      data-nodeid="1"
    />
    <div
      class="bracket-node__title "
    >
      <div
        class="bracket-node__name "
      >
        None
      </div>
    </div>
  </div>
</DocumentFragment>
`;
